feat: 파일 업로드 스타터 구현

hyeblee opened the pull request:
## 이슈 번호

close #4

## 작업 사항
- `ftp-starter` 모듈 추가 및 파일 관련 로직 스타터로 분리
- 파일 업로드, 삭제, 존재 여부 확인, 저장 주소 반환 기능 구현
- `communication-service`에 FtpController 추가

## 스크린샷 (선택)

---

## 공유사항
<ftp 사용 방법>
- `application-ftp.yml` 파일 추가 필요
- dependencies에 아래 의존성 추가 필요
```java
implementation project(':libs:ftp-starter')
```

- 스타터 빈 자동등록 실패 문제로, 사용 서비스에 아래 어노테이션 필요
```java
@SpringBootApplication(scanBasePackages = { "com.hermes.ftpstarter" })
```

hyeblee committed:
feature: 공통 응답 클래스 생성

hyeblee committed:
feat: FTP 스타터 초기 구현

- 파일 업로드, 삭제, 존재 확인, 저장 URL 반환 기능 구현
- UUID 기반 파일명 고유화 적용
- 업로드 성공 시 저장된 파일명만 반환
- FTP 스타터 사용 시 application-ftp.yml 추가 필요
- 사용하는 서비스에서 @Autowired로 FtpService 사용 가능
- ApiResponse는 추후 라이브러리 구현 시 수정 후 삭제 예정

hyeblee committed:
feat: 커뮤니케이션 서비스 초기 설정 및 FTP API 구현

- FTP 관련 컨트롤러 구현
- 정상 동작을 위해 application-communication-db.yml, application-ftp.yml 필요
- FtpController에서 실패 시 예외 처리 로직 추가 필요

hyeblee committed:
refactor: 불필요한 scanBasePackages 제거 및 스타터 모듈 설정 추가

- ftp-spring-boot-starter는 자동 빈 등록이 되므로 CommunicationServiceApplication에서 scanBasePackages 설정을 제거
- settings.gradle에 communication-service와 ftp-spring-boot-starter 모듈 포함 설정

hyeblee committed:
fix: communication-service에서 FtpService 빈 주입 위해 컴포넌트 스캔 유지

- @SpringBootApplication(scanBasePackages = {...}) 어노테이션 제거 시 스타터 빈 등록 실패
- 스타터 자동등록 사용을 시도했으나, 문제 발생으로 기존 스캔 유지

hyeblee committed:
chore: settings.gradle에 communication-service와 ftp-spring-boot-starter 추가

hyeblee committed:
Merge branch 'develop' into feature/4-ftp-upload

Othereum reviewed (changes requested):
ftp-spring-boot-starter -> libs/ftp-starter
모듈 위치 이동 및 이름 변경 부탁드립니다

hyeblee committed:
refactor: ftp-starter 이름 수정 및 위치 이동

hyeblee committed:
Merge branch 'feature/4-ftp-upload' of https://github.com/backend20250319/BE09-Final-4team-BE into feature/4-ftp-upload

hyeblee committed:
refactor: ftp-starter 패키지명 변경

- 기존 com.hermes.ftpspringbootstarter -> com.hermes.ftp-starter로 변경
- 스타터 내부 클래스, 패키지 임포트 경로 모두 수정
- 멀티모듈 구조에서 일관성 있는 패키지명 적용

hyeblee committed:
refactor: ApiResponse를 공통 라이브러리로 변경 및 파일 URL 반환 메서드 오류 수정
- 파일 이름을 반환하던 메서드를 다운로드 가능한 url 반환하게 수정

hyeblee commented:
@Othereum 해당 부분 수정했습니다 확인 부탁드려요!

Othereum commented on settings.gradle:
```diff
@@ -18,3 +18,6 @@ include 'leave-service'
 
+include 'communication-service'
+include 'libs:ftp-starter'
+project(':libs:ftp-starter').projectDir = new File(rootDir, 'libs/ftp-starter')
```
23번줄 필요한건가요?

hyeblee replied:
해당 부분 수정하여 다시 올렸습니다!

hyeblee committed:
refactor: 불필요한 프로젝트 경로 지정 제거
